lib/wda/xcodebuild.js

```diff
@@ -106,11 +107,17 @@ class XcodeBuild {
       }
       // now parse the log file for the derived data folder
       const logFile = match[1];
-      stdout = await fs.readFile(logFile);
-      match = DERIVED_DATA_FOLDER_REGEXP.exec(stdout);
-      if (!match) {
-        // nothing found. We are done
-        log.debug(`Cannot find the derived data location from the xcodebuild log file '${logFile}'`);
+      try {
+        // grep on the log file, since it might be too big to fit it completely into the memory
+        const grepData = await exec('grep', DERIVED_DATA_GREP_REGEXP, logFile);
```

Comment:
`exec` takes a command and an array of arguments to that command. https://github.com/appium/node-teen_process/blob/master/lib/teen_process.js#L11

Reply:
fixed